normalized, filing and prefix token types

codeforkjeff opened the pull request:
JsonReferencePayloadTokenizer now emits normalized, filing and prefix token types, and these are included in facets

magibney commented on src/main/java/edu/upenn/library/solrplugins/JsonReferencePayloadHandler.java:
```diff
@@ -12,33 +11,81 @@
+  /**
+   * Copies a field value from one NamedList into another
+   */
+  private static void copyField(NamedList<Object> from, NamedList<Object> to, String key) {
```
"copyField" is [overloaded in a Solr context](https://cwiki.apache.org/confluence/display/solr/Copying+Fields) ... consider changing this method name to avoid confusion?

codeforkjeff replied:
Cool, I've renamed it to copyFieldInNamedList. While I was at it, there's a bunch of places that look for a key in a NamedList whose value is another Named List, creating one if it doesn't exist; I refactored these into a getOrCreateNamedListValue().

codeforkjeff commented:
I've made the changes we talked about at yesterday's meeting: "normalized" token is now created by a filter instead of the tokenizer; we don't need to store the normalized value or return it as part of the facet payload; added a "self" key to the facet payload; fixed bug where "refs" key was missing in facet payload

codeforkjeff commented:
Oops, I forgot to modify mergePayload() to account for 'self'. I've added this to the PR.
